tensorflow/lite/experimental/swift/BUILD.apple

```diff
@@ -25,18 +25,47 @@ package(
 
 swift_library(
     name = "TensorFlowLite",
-    srcs = glob(["Sources/*.swift"]),
-    linkopts = [
-        "-Wl,-weak_framework,CoreML",
-        "-Wl,-weak_framework,Metal",
-    ],
+    srcs = glob(
+        [
+            "Sources/*.swift",
+        ],
+        exclude = [
+            "Sources/CoreMLDelegate.swift",
+            "Sources/MetalDelegate.swift",
+        ],
+    ) + select({
+        "//tensorflow/lite/tools/delegates:use_coreml_delegate": [
+            "Sources/CoreMLDelegate.swift",
+        ],
+        "//conditions:default": [],
+    }) + select({
+        "//tensorflow/lite/tools/delegates:use_metal_delegate": [
+            "Sources/MetalDelegate.swift",
+        ],
+        "//conditions:default": [],
+    }),
+    linkopts = select({
+        "//tensorflow/lite/tools/delegates:use_coreml_delegate": [
+            "-Wl,-weak_framework,CoreML",
+        ],
+        "//conditions:default": [],
+    }) + select({
+        "//tensorflow/lite/tools/delegates:use_metal_delegate": [
+            "-Wl,-weak_framework,Metal",
+        ],
+        "//conditions:default": [],
+    }),
     module_name = "TensorFlowLite",
     tags = TFL_DEFAULT_TAGS,
     visibility = ios_visibility_whitelist(),
     deps = [
-        "//tensorflow/lite/experimental/delegates/coreml:coreml_delegate",
         "//tensorflow/lite/experimental/ios:tensorflow_lite_c",
-    ],
+    ] + select({
```

Comment:
Similar to comment for `TensorFlowLiteC_framework`, we will want Metal to follow same pattern as CoreML. I think @teijeong is planning to update Metal to match CoreML?

Probably out of scope for your current change, so seems fine to leave as you have it and we can add Metal to this list of deps when ready.

Reply:
The metal cocoapods changes were merged yesterday, I've rebased to work with the new pattern